lib/core/src/Data/Quantity.hs

```diff
@@ -134,49 +135,65 @@ instance (KnownSymbol unit, Buildable a) => Buildable (Quantity unit a) where
 
 -- | Opaque Haskell type to represent values between 0 and 100 (incl).
 newtype Percentage = Percentage
-    { getPercentage :: Word }
+    { getPercentage :: Scientific }
     deriving stock (Generic, Show, Eq, Ord)
     deriving newtype (ToJSON)
 
+-- | Desired number of digits after the decimal point for presenting the
+-- @Percentage@ type.
+percentageNumberOfFractionalDigits :: Int
+percentageNumberOfFractionalDigits = 2
+
 instance NFData Percentage
 
 instance FromJSON Percentage where
     parseJSON x = do
-        n <- parseJSON x
-        either (fail . show) return (mkPercentage @Int n)
+        r <- parseJSON x
+        either (fail . show) return (mkPercentage r)
 
 instance Bounded Percentage where
     minBound = Percentage 0
     maxBound = Percentage 100
 
-instance Enum Percentage where
-    fromEnum (Percentage p) = fromEnum p
-    toEnum = either (error . ("toEnum: " <>) . show) id . mkPercentage
-
 instance ToText Percentage where
-    toText (Percentage p) = T.pack (show p) <> "%"
+    toText (Percentage p) = T.pack (showS p) <> "%"
+      where
+        showS =
+            formatScientific
+            Fixed
+            (Just percentageNumberOfFractionalDigits)
 
 instance FromText Percentage where
     fromText txt = do
-        (p, u) <- left (const err) $ decimal txt
+        (p, u) <- left (const err) $ rational txt
         unless (u == "%") $ Left err
-        left (const err) $ mkPercentage @Integer p
+        left (const err) . mkPercentage $ (p / 100)
       where
         err = TextDecodingError
             "expected a value between 0 and 100 with a '%' suffix (e.g. '14%')"
 
 -- | Safe constructor for 'Percentage'
 mkPercentage
-    :: Integral i
-    => i
+    :: Double
     -> Either MkPercentageError Percentage
-mkPercentage i
-    | let Percentage inf = minBound in i < fromIntegral inf =
+mkPercentage r
+    | r < 0 =
         Left PercentageOutOfBoundsError
-    | let Percentage sup = maxBound in i > fromIntegral sup =
+    | r > 1 =
         Left PercentageOutOfBoundsError
     | otherwise =
-        pure $ Percentage $ fromIntegral i
+        let
+            toPercent = (* 100)
+        in
+            pure . Percentage . round' . toPercent $ r
+  where
+    round' :: Double -> Scientific
+    round' x = (fromRational (toRational i)) / (fromRational (toRational factor))
+      where
+        i :: Int
+        i = round (factor * (toRational x))
+
+        factor = 10 ^ percentageNumberOfFractionalDigits
```

Comment:
This function has several issues here:

a) It forces `Double` as an argument, which pretty much looses all the benefits of using a type like Scientific or Ratio underneath.

b) It mixes formatting with the construction of the type by already rounding the argument. 

Why not choose `Rational` for the inner type of the `Percentage` type and do the formatting / cast to 2-decimal scientific only at the API boundaries?

Reply:
> which pretty much looses all the benefits of using a type like Scientific or Ratio underneath.

Firstly: What benefits? Any potential errors would be hidden by the rounding?

Secondly: Things get more complicated with `Rationals`.

```haskell
        ActiveSlotCoefficient f = sp ^. #getActiveSlotCoefficient
        remainingBlocks = remainingSlots * f
        progress = bhTip % (bhTip + remainingBlocks)
```
This is essentially `a/(b+c) where, a:Word, b:Word, c:Double`.

What do we do?
- Do the calculation as a `Double` and _then_ convert to `Rational`? Seems iffy to me.
- Round `remainingBlocks`? I guess this will make the sync progress less smooth.

Reply:
> Why not choose Rational for the inner type of the Percentage type

Maybe we could use `Scientific` 🤔

Reply:
> Firstly: What benefits? Any potential errors would be hidden by the rounding?

PRECISION! Double are not good at representing rational numbers. Yet, percentage are almost always computed as a ratio of two numbers, hence why `Rational` is a better fit here.

> Secondly: Things get more complicated with Rationals.

This sounds actually much simpler to me... We can safely round `remainingBlocks` to an integer actually, because it's a number of block! What does it mean to have 3.4 remaining blocks anyway? Flooring or Ceiling doesn't really matter here because this is a progress reporting, not something that need more careful thinking. 

> Round remainingBlocks? I guess this will make the sync progress less smooth.

We are talking about 4M blocks for mainnet and ~2.5M for the ITN. Really, a 1 block error margin is fine.

Reply:
> This sounds actually much simpler to me

I'm happy there's an approach you like 🙏 

Still, for the sake of it:

> Double are not good at representing rational numbers. 

Seems like a vague and general argument.

> Yet, percentage are **almost always** computed as a ratio of two numbers, hence why 

Rational is a better fit here.

In this case we had an instance of `(a/(b+(c/d)))` (basically) as you saw. 

- Using `round` might introduce an error of `2e-6` (on [0,1] sync progress for ITN).*
- We are displaying `Percentage` with a precision of `1e-4` (i.e. 100.00%=1.0000). 
- The error we get from a `Double` must surely be smaller than **even** 1e-14?

The 2e-6 should _probably_ not matter. But 1e-14 should matter even less.

> What does it mean to have 3.4 remaining blocks anyway?

It _might_ be useful to make the sync progress smoother for small values of `f`, in some circumstances. If f=1%, should the progress only start decreasing once we are 100 slots behind the current time? 🤷‍♂ 

*)
According to pooltool.io the block height is ≈273000.
If we use `round` the average error should be 0.5.
The error in syncProgress is then 0.5/273000 = 2e-6

_modulo minor math notation / terminology mistakes_

Reply:
> Seems like a vague and general argument in these circumstances.

Not really........

> In computing, floating-point arithmetic (FP) is arithmetic using formulaic representation of real numbers as an approximation to support a trade-off between range and precision. 

source: https://en.wikipedia.org/wiki/Floating-point_arithmetic

I mean, that's simple enough as an explanation. Floating point arithmetic is a vast topic in computer science but the basic idea is easy to grasp: floating numbers are an approximation. 

Now, what I am saying is not that the approximation in this particular case is not okay (as you just outlined), but that if we can avoid this approximation by actually using a type that enables us to represent values as non-approximated, we are better off using that approach. This actually saves us a lot of headache and hours trying to calculate and demonstrate how rounding this particular value is okay. Using a Rational, we only have to worry about the moment we need to convert it to a float and chose a given precision to work with. Instead of carrying around an approximation error where it's not needed.

Reply:
I think you missed the point of my comment here... From the first comment: 

> Why not choose Rational for **the inner type of the Percentage type** [...]

Now, we have a Rational as an argument that we convert to double and convert back to Scientific!